### PR TITLE
Add dividends, member count and token supply

### PR DESCRIPTION
**Pull request by agentMAK**
There are 3 main new entity types. 
`IndieMemberCount` - which tracks the active, claimed, terminated members, etc. 
`TokenSupply`- which tracks the total Indie token supply
`DividendProfit` - which tracks the dividends and net profit for the DAO

Each of the entities is accessible via the `quarters `entity, such that historical data can be accessed

### Implementation details
- For the `IndieMemberCount` entity, the smart contract events for claimed, active, inactive, terminated, and resigned are tracked. The `active` entity is based on a count, as it can't be accessed directly. So we use `+1` for active events and `-1` for terminated, resigned and inactive events. The active entity currently assumes that a member is active when they are terminated or resigned. The alternative would be to store the addresses of every member in the subgraph.
- For `TokenSupply`, we currently track each transfer for mint/burns and update the token supply accordingly
- For the `dividends` we call the smart contract, via the dividendsBySeason read function to divide by the total members.
- Finally, each of the entities is updated via their respective events and the transfer event, to cover the case where there are no changes to the values and new quarter values hasn't been set.

### CODE REVIEW

File: src/quarter.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { Quarter } from "../generated/schema";
+import { BigInt } from "@graphprotocol/graph-ts";
+
+export function _findOrCreateQuarter(year: i32, quarter: i32): Quarter {
```
Comment (seanconnollydev) [inline src/quarter.ts]: We can drop the `_` on these functions if we're exporting them and making them public.

File: tests/indie-token.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import {
+  assert,
+  describe,
+  test,
+  clearStore,
+  beforeAll,
+  afterAll
+} from "matchstick-as/assembly/index"
+import { Address, BigInt, Bytes } from "@graphprotocol/graph-ts"
+import { createAdminChangedEvent } from "./indie-token-utils"
```
Comment (seanconnollydev) [inline tests/indie-token.test.ts]: I would like to get some tests in here eventually but we can remove this if we're not using it.

File: src/indie-token.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+
+export function handleTransfer(event: TransferEvent): void {
+  _updateTokenSupply(event);
+
+  _updateTotalMemberCount();
+
+  _updateQuarterlyIndieMemeberCount(event.block.timestamp);
+
+  _updateQuarterlyDividend(event)
```
Comment (seanconnollydev) [inline src/indie-token.ts]: I bet there is a better event to update the quarterly dividend values. I believe we just need to know what the amount was for the season and the number of indies. @mckelvey should be able to help us here. Maybe it's when we call `createSeasonSnapshot`?

**Review by seanconnollydev — COMMENTED** [review]
This is looking solid @agentMAK! Just a few minor adjustments I think.

File: src/indie-token.ts (hunk 3 above)
Comment (mckelvey) [reply src/indie-token.ts]: The on chain event name is `SeasonalDividend`:

```
    /**
     * @dev Emitted when a seasonal USDC dividend total has been set
     */
    event SeasonalDividend(uint256 indexed seasonId, uint256 totalDividend, uint256 totalWithholding);
```

For this case I’d assume you could track `SeasonalDividend`... Just note that seasonId index 1 is what we call Season 7, or 2023 Q2, July through Sept. (Seasons 1-6 were the original $INDIE token, which was a separate contract.) Season 8, 2023 Q3, our most recent completed season is index `2`.

For the someday when a dash like this is member-specific, just know that there is also this event `SeasonalMemberDividend`:

```
    /**
     * @dev Emitted when a seasonal USDC dividend has been assigned to an indie member
     */
    event SeasonalMemberDividend(
        uint256 indexed seasonId, address indexed memberAddress, uint256 netDividend, uint256 withholding
    );
```

**Review by seanconnollydev — APPROVED** [review]
This looks great @agentMAK! Nice work!

**Pull request merged**